docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,63 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, namespace) to individual log entries, and sending the logs to the Elasticsearch cluster. The pod metadata is sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  This results in logs not being shipped to the aggregator and thus being unavailable.
```

Comment:
Sorry? 1.5 will only support http2? Can you link to github issues for that please? There are clients in many languages that will simply not be able to migrate to http2 yet AFAIK.

> B/c the watches are are being done outside of the core client I'm wondering what level of connection reuse exists.

This uses the same Ruby client that ManageIQ uses (https://github.com/abonas/kubeclient) which in turn uses the http gem (https://github.com/httprb/http). This seems to support connection reuse so...

> During last experiments, we exhausted ports on our LB @ 500 Nodes.

Could you capture number of connections per node please? Also running something on a node to capture how many connections fluentd is making would be great too if possible.